### DISCUSSION TITLE
signinKey for ImageUpdateAutomation should be CrossNamespace

### DISCUSSION THREAD

**Discussion by ViBiOh** [fluxcd/flux2#2383] [Proposals] [answered]
Hi there 👋 

In the latest flux release, the [Image Update Automation](https://fluxcd.io/docs/components/image/imageupdateautomations/#cross-namespace-references) can now refer a `GitRepository` in another namespace, that's very cool to avoid duplicating ssh credentials accross namespace 👍 

One last mile, from my point of view, is to allow the `SecretRef` in the Commit's SigninKey to be also CrossNamespace (to point the very same credentials of the git repository). What do you think about it ?

My final "dreamed image automation" will the one I described [here](https://github.com/fluxcd/image-automation-controller/issues/85#issuecomment-788933978) (still in discussion, and I'm okay with that):
* one repository with one credential at one place
* image policies in various namespaces
* one automation, in the same namespace than the repository, that grabs every policy it founds in a given repository (based on the comment added to yaml).

**Comment by stefanprodan** [comment, marked as answer]
The Flux API has no cross-namespace support for referencing Secrets or ConfigMaps. I don't expect this to change, unless Kubernetes allows cross-namespace access to secrets in its own API, which will probably never happen.

**Reply by ViBiOh** [reply]
Legit, thank you for your answer ;)
